python-threatexchange/threatexchange/extensions/vpdq/README.md

```diff
@@ -0,0 +1,85 @@
+# Summary
+The description of vPDQ algorithm hashing and brute-force matching can be found in [vpdq/README.md](../../../../vpdq/README.md). This README describes the [FAISS](https://github.com/facebookresearch/faiss) matching process which is integrated with the vPDQ index in python-threatexchange.
+
+#### Faiss matching
+vPDQ features are a list of PDQ hashes embedded with Frame, Quality, and Timestamp(sec). When added to FAISS index, only PDQ hashes are indexed. Thus, extra information is stored in the wrapper index class to reformat the vPDQ feature when returning the results for searching. When we want to match a video (a list of vPDQ features) against the index, we also only input the PDQ hashes to search.
+
+A query on the index is comprised of a sequence of individual PDQ hashes. Each of which will have a corresponding set of matches returned that were within threshold hamming distance of that query (or none in the case that specific hashes did not match anything in the index). These matches will be (idx, distance).
+The inner sequences may be empty in the case of no hashes within the index.
+The same VPDQ feature may also appear in more than one inner sequence if it matches multiple query hashes.
+For example the hash "000000000000000000000000000000000000000000000000000000000000ffff" would match both
+"00000000000000000000000000000000000000000000000000000000fffffff" and
+"0000000000000000000000000000000000000000000000000000000000000000" for a threshold of 16. Thus it would appear in
+the entry for both the hashes if they were both in the queries list.
+
+e.g.
+query_str =>  (idx, distance)
+result = {
+    "000000000000000000000000000000000000000000000000000000000000ffff": [
+        (12345678901, 16.0)
+    ]
+}
+
+To calculate the percentage of query matching and index matching (i.e. what percentage of frames in the search video query appear in the index and vice versa), two unique sets are used in the wrapper class to count the number of matches.
+
+The query video’s frame PDQ hashes Q
+Two unique comparison videos' frame PDQ hashes C1,C2
+The PDQ match distance D (example: 31)
+The PDQ quality filter tolerance F (example: 50), if either hash is below this quality level then they will not be compared
+Here is the algorithm, in pseudocode:
```

Comment:
nit/ignorable: this is more of a process or using and index than an algorithm

Reply:
How to use index "algorithm"